examples/applications/object_detection/retina_net/basic/pascal_voc/loss.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+
+
+class FocalLoss(tf.losses.Loss):
+    def __init__(self, num_classes, alpha=0.25, gamma=2.0, delta=1.0, **kwargs):
+        super().__init__(**kwargs)
+        self.num_classes = num_classes
+        self.classification_loss = RetinaNetClassificationLoss(
+            alpha=alpha,
+            gamma=gamma,
+        )
+        self.box_loss = RetinaNetBoxLoss(delta=delta)
+
+    def call(self, y_true, y_pred):
+        box_labels = y_true[:, :, :4]
+        box_predictions = y_pred[:, :, :4]
+
+        cls_labels = tf.one_hot(
+            tf.cast(y_true[:, :, 4], dtype=tf.int32),
+            depth=self.num_classes,
+            dtype=tf.float32,
+        )
+        cls_predictions = y_pred[:, :, 4:]
+        positive_mask = tf.cast(tf.greater(y_true[:, :, 4], -1.0), dtype=tf.float32)
```

Comment:
It seems that we have some strong assumption not only for shape, but also for value in the bbox. We might want to put this in the docstring.

Reply:
https://github.com/keras-team/keras-cv/pull/615